cmd/kms-rest/startcmd/start.go

```diff
@@ -188,3 +209,100 @@ func healthCheckHandler(rw http.ResponseWriter, r *http.Request) {
 		log.Errorf("healthcheck response failure, %s", err)
 	}
 }
+
+type keystoreProvider struct {
+	storageProvider storage.Provider
+	kmsCreator      provider.KMSCreator
+	crypto          crypto.Crypto
+}
+
+func (k keystoreProvider) StorageProvider() storage.Provider {
+	return k.storageProvider
+}
+
+func (k keystoreProvider) KMSCreator() provider.KMSCreator {
+	return k.kmsCreator
+}
+
+func (k keystoreProvider) Crypto() crypto.Crypto {
+	return k.crypto
+}
+
+type kmsProvider struct {
+	storageProvider ariesstorage.Provider
+	secretLock      secretlock.Service
+}
+
+func (k kmsProvider) StorageProvider() ariesstorage.Provider {
+	return k.storageProvider
+}
+
+func (k kmsProvider) SecretLock() secretlock.Service {
+	return k.secretLock
+}
+
+func createKeystoreProvider(parameters *kmsRestParameters) (provider.Provider, error) {
+	c, err := tinkcrypto.New()
+	if err != nil {
+		return nil, err
+	}
+
+	return keystoreProvider{
+		storageProvider: memstore.NewProvider(),
+		kmsCreator:      prepareKMSCreator(parameters),
+		crypto:          c,
+	}, nil
+}
+
+func prepareKMSCreator(_ *kmsRestParameters) provider.KMSCreator {
+	return func(keystoreID string) (kms.KeyManager, error) {
+		kmsStorageProv := ariesmemstorage.NewProvider()
+
+		masterKeyReader, err := prepareMasterKeyReader(kmsStorageProv)
+		if err != nil {
+			return nil, err
+		}
+
+		secretLockService, err := local.NewService(masterKeyReader, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		kmsProv := kmsProvider{
+			storageProvider: kmsStorageProv,
+			secretLock:      secretLockService,
+		}
+
+		localKMS, err := localkms.New(fmt.Sprintf(masterKeyURI, keystoreID), kmsProv)
+		if err != nil {
+			return nil, err
+		}
+
+		return localKMS, nil
+	}
+}
+
+func prepareMasterKeyReader(kmsStorageProvider ariesstorage.Provider) (*bytes.Reader, error) {
+	masterKeyStore, err := kmsStorageProvider.OpenStore(masterKeyStoreName)
+	if err != nil {
+		return nil, err
+	}
+
+	masterKey, err := masterKeyStore.Get(masterKeyDBKeyName)
+	if err != nil {
+		if errors.Is(err, ariesstorage.ErrDataNotFound) {
+			const size = 32
+			masterKeyRaw := random.GetRandomBytes(uint32(size))
```

Comment:
use crypto.Rand here

Reply:
did you mean using `rand.Read` from the standard `crypto/rand` package as in snippets below?

```go
func generateRandomBytes() []byte {
	keySize := sha256.New().Size()
	buf := make([]byte, keySize)

	_, err := rand.Read(buf)
	if err != nil {
		panic(err) // out of randomness, should never happen :-)
	}

	return buf
}
```

Reply:
yes

Reply:
done cc @Baha-sk